Define a Cargo workspace

not-my-profile opened the pull request:
Just makes development a bit more comfortable, e.g. `cargo run` now works in the repository root directory ... and you can now open the repository in VS Code and rust-analyzer will work also for the other non-root crates.

not-my-profile commented on Cargo.toml:
```diff
@@ -1,3 +1,7 @@
+[workspace]
+members = [".", "cargo-insta", "cargo-insta/integration-tests"]
+default-members = [".", "cargo-insta"]
```
This `default-members` setting is what enables `cargo run` to work in the repository root.

mitsuhiko commented:
I know this is really appealing, and I really want to do this. Unfortunately this forces all dependency resolution to be shared which breaks the test runs on older Rust versions :(

not-my-profile commented:
Ah yes compilation indeed failed but only because I deleted the `Cargo.lock` file making Cargo pick some dependencies that used Rust edition 2021, which isn't yet available in our MSRV 1.51.0.

I have now added the `Cargo.lock` file of the workspace to the commit ... I did have to run the following five commands (basically downgrading to the previous dependency version everytime the error came up):

```
cargo update -p once_cell --precise 1.14.0
cargo update -p pest_derive --precise 2.3.1
cargo update -p pest_generator --precise 2.3.1
cargo update -p pest_meta --precise 2.3.1
cargo update -p pest --precise 2.3.1
```

But the crate now builds with Rust 1.51.0 :) (Note that `cargo test` still fails but that's also happening on master because of features used in some `tests/*` files).

mitsuhiko commented:
Yeah, i'm aware of this but I don't like that setup. I know it's not a lot of fun with two workspace and eventually I want to fix this, but for now I prefer the current setup.

not-my-profile commented:
I am curious, what do you dislike about that setup?

max-sixty commented:
Would a reasonable compromise be accepting a PR when the MSRV catches up?

mitsuhiko commented:
@not-my-profile I do not want to require specific lock files for tests to pass. That makes it too easy to miss when dependencies on the insta crate create an MSRV incompatibility. That's why I want to have these separated for now. Insta should be install and testable on its own MSRV rules without a lockfile but that has different requirements than cargo-insta.

@max-sixty that's my goal, yes.
